### PR TITLE
Adding app/mongoose project options and deprecating connect()

### PR DESCRIPTION
**Pull request by JohnnyEstilles**
Fixing issue #1036

### CODE REVIEW

File: lib/core/connect.js [hunk 1]
```diff
@@ -14,6 +14,12 @@
 var debug = require('debug')('keystone:core:connect');
 
 function connect() {
+	var warningMsg = 'keystone.connect() is now deprecated and will not be available in future versions of KeystoneJS\n' +
```
Comment (JedWatson) [inline lib/core/connect.js]: I think we should actually throw here, since the `connect` function won't work at all in 0.3

**Comment by JedWatson** [comment]
For consistency we should also make sure that `init()` uses `set` too, otherwise `get('express')` won't always work.

```
	if (!this.app) {
		this.set('express'), require('express')());
	}
	
	if (!this.mongoose) {
		this.set('mongoose', require('mongoose'));
	}
```

File: lib/core/connect.js (hunk 1 above)
Comment (JohnnyEstilles) [inline lib/core/connect.js]: Agreed!

**Comment by JohnnyEstilles** [comment]
Agreed! Will push commits in a few ...

File: lib/core/connect.js (hunk 1 above)
Comment (JohnnyEstilles) [inline lib/core/connect.js]: I'll updated the deprecation warning in docs as well.

**Comment by JohnnyEstilles** [comment]
Done!

**Comment by JedWatson** [comment]
:+1:

**Pull request merged**